File: tests/r27b1b/test_no_backend_inference.ts
import test from "node:test";
import assert from "node:assert/strict";
import { readFileSync, readdirSync } from "node:fs";
import { join, resolve } from "node:path";

test("R27B1B runtime files contain no backend inference or remote LLM endpoints", () => {
  const root = resolve(process.cwd());
  function runtimeFiles(dir) {
    return readdirSync(dir, { withFileTypes: true }).flatMap((entry) => {
      const path = join(dir, entry.name);
      if (entry.isDirectory()) return runtimeFiles(path);
      return path.endsWith(".ts") ? [path] : [];
    });
  }
  const files = [
    ...runtimeFiles(join(root, "src/browser_runtime")),
    join(root, "web/another_brain_chat/browser_runtime.js"),
    join(root, "web/another_brain_chat/runtime_worker.js")
  ];
  const text = files.map((file) => readFileSync(file, "utf8")).join("\n");
  assert.doesNotMatch(text, /FastAPI|Flask|app\.post|api\.openai\.com|anthropic\.com|doubao|dashscope|volces/i);
});

File: tests/r27b1a/test_no_backend_inference.py
import unittest

from pathlib import Path


ROOT = Path(__file__).resolve().parents[2]


class R27B1ANoBackendInferenceTests(unittest.TestCase):
    def test_r27b1a_scripts_do_not_define_backend_inference(self):
        for path in sorted((ROOT / "scripts").glob("r27b1a_*.py")):
            text = path.read_text(encoding="utf-8")
            self.assertNotIn("FastAPI", text)
            self.assertNotIn("Flask", text)
            self.assertNotIn("app.post", text)
            self.assertNotIn("api.openai.com", text)
            self.assertNotIn("doubao", text.lower())


if __name__ == "__main__":
    unittest.main()
